openstudiocore/src/model_editor/CMakeLists.txt

```diff
@@ -34,9 +34,8 @@ set(${target_name}_src
   OSProgressBar.cpp
   PathWatcher.hpp
   PathWatcher.cpp
-  #  TODO: Unsure if needed? 
   QMetaTypes.hpp
-  QMetaTypes.cpp
+  # QMetaTypes.cpp # Not needed
```

Comment:
@jmarrec how did you determine that this source file was not needed, just that there are no linker errors?  this code gets called at run time, I think the symptoms would be parts of the app not working

Reply:
I tried again this morning the OS App, and I couldn't see any problems without the Cpp, though I do think we *should* need them indeed.

From the [Qt documentation](https://doc.qt.io/qt-5/qmetatype.html#Q_DECLARE_METATYPE):

> Adding a Q_DECLARE_METATYPE() makes the type known to all template based functions, including QVariant. Note that if you intend to use the type in **queued** signal and slot connections or in QObject's **property** system, you also have to call qRegisterMetaType() since the names are resolved at runtime.

so `Q_PROPERTY` is gone now, so not a problem. The question stays whether we have our custom types using as **arguments** inside a **queued** signal and slot connection.

I think we do... cf `ModelObjectListsView.hpp`:


```c++
class ModelObjectListController : public OSVectorController
[...]
private slots:
 void objectAdded(std::shared_ptr<openstudio::detail::WorkspaceObject_Impl>, const openstudio::IddObjectType&, const openstudio::UUID&);
```

`ModelObjectListsView.cpp`:


```c++
 connect(OSAppBase::instance(), &OSAppBase::workspaceObjectAddedPtr, this, &ModelObjectListController::objectAdded, Qt::QueuedConnection)

```

Anyways,I re-added the qRegisterMetaType calls inside a static initializer (and put the cpp file into CMakeLists again)

Reply:
Last commit is an alternate way, which doesn't use the initializer Struct, I think you prefer this one @macumber but if not you can revert the commit